### DISCUSSION TITLE
Recent TechEmpower benchmark strange result

### DISCUSSION THREAD

**Discussion by aentwist** [micronaut-projects/micronaut-core#8585] [General] [answered]
Why is Micronaut coming in at way below even Grails in the recent TechEmpower benchmark?

See https://www.techempower.com/benchmarks/#section=data-r21

**Comment by graemerocher** [comment, marked as answer]
See https://github.com/TechEmpower/FrameworkBenchmarks/discussions/7618

**Reply by aentwist** [reply]
Damn that sucks. Y'all got straight up sabotaged 😂 I hope the benchmark maintainers work with you on that, or the benchmark's credibility will be vastly undermined in my eyes.

**Reply by yawkat** [reply]
@aentwist unfortunately TE benchmarks aren't often representative, they depend heavily on how much time people actually spend optimizing a particular test case. This is easy to see e.g. with vertx vs netty: Vertx is *based on* netty, so it can't really be faster than netty, but it still beats netty in some of the TE benchmarks.
